pkg/awsds/asyncDatasource_test.go

```diff
@@ -103,3 +104,41 @@ func Test_getDBConnectionFromQuery(t *testing.T) {
 		})
 	}
 }
+
+func TestAsyncQueryData(t *testing.T) {
```

Comment:
I'd suggest explaining more what the behavior is in the test title and/or the `desc` cases.

```suggestion
func Test_Async_QueryData_uses_synchronous_flow_when_header_has_alert_and_expression(t *testing.T) {
```

Reply:
done